docs/csharp/misc/cs0459.md

```diff
@@ -1,40 +1,45 @@
 ---
 title: "Compiler Error CS0459"
 ms.date: 07/20/2015
-f1_keywords: 
+f1_keywords:
   - "CS0459"
-helpviewer_keywords: 
+helpviewer_keywords:
   - "CS0459"
 ms.assetid: 01b058dd-8d65-4e9d-9de1-d47f9488d22a
 ---
 # Compiler Error CS0459
-Cannot take the address of a read-only local variable  
-  
- There are three common scenarios in the C# language that generate read-only local variables: `foreach`, `using`, and `fixed`. In each of these cases, you are not allowed to write to the read-only local variable, or to take its address. This error is generated when the compiler realizes you are trying to take the address of a read-only local variable.  
-  
-## Example  
- The following example generates CS0459 when an attempt is made to take the address of a read-only local variable in a `foreach` loop and in a `fixed` statement block.  
-  
-```csharp  
-// CS0459.cs  
-// compile with: /unsafe  
-  
-class A  
-{  
-    public unsafe void M1()  
-    {  
-        int[] ints = new int[] { 1, 2, 3 };  
-        foreach (int i in ints)  
-        {  
-            int *j = &i;  // CS0459  
-        }  
-  
-        fixed (int *i = &_i)  
-        {  
-            int **j = &i;  // CS0459  
-        }  
-    }  
-  
-    private int _i = 0;  
-}  
+
+Cannot take the address of a read-only local variable
+
+ There are three common scenarios in the C# language that generate read-only local variables: `foreach`, `using`, and `fixed`. In each of these cases, you are not allowed to write to the read-only local variable, or to take its address. This error is generated when the compiler realizes you are trying to take the address of a read-only local variable.
+
+## Example
+
+ The following example generates CS0459 when an attempt is made to take the address of a read-only local variable in a `foreach` loop and in a `fixed` statement block:
+
+```csharp
+// CS0459.cs
+// compile with: /unsafe
+
+class Program
+{
+    public unsafe void M1()
+    {
+        int[] ints = new int[] { 1, 2, 3 };
+        foreach (int i in ints)
+        {
+            int *j = &i;  // CS0459
+        }
+
+        fixed (int *i = &_i)
+        {
+            int **j = &i;  // CS0459
+        }
+    }
+
+    private int _i = 0;
+}
 ```
+
+> [!NOTE]
+> Roslyn compiler was updated and this compiler error was removed starting with Visual Studio 2017 15.5, so the previous code would compile successfully with this version and later.
```

Comment:
What if someone uses Visual Studio Code? In other words, is it possible to indicate the C# version, starting with which the example compiles?

Reply:
I really was thinking on that while writing it, but have no idea.

Reply:
It looks that Visual Studio 2017 15.5 was shipped with C# 7.2. However, if I use one of the latest .NET Core SDKs and set the language version to 7.0 or 6.0, the code still compiles. So, indeed, it's more about the Roslyn version and not the C# version.

Reply:
Based on [this list](https://github.com/dotnet/roslyn/wiki/NuGet-packages#versioning), VS 15.5 corresponds to Roslyn 2.6.  Though I think most people are not aware of the version of Roslyn they use, so that information is probably not useful.

Based on [this page](https://dotnet.microsoft.com/download/dotnet-core/2.0), C# 7.2/VS 15.5 corresponds to .Net Core SDK 2.1.100. Maybe that's a version that would be worth mentioning?

Reply:
@svick, The second page says that VS 15.5 supports C# 7.2, this doesn't mean that the error will be generated before C# 7.2, because I can be using VS 15.5 with some older C# version.

Not sure if Roslyn 2.6 should be mentioned or not.